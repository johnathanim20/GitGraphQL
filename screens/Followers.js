import React, { useState, useEffect} from 'react';
import { Text, View, Button } from 'react-native';
import {parseProfile} from '../parsers/parseProfile.js';
import { API_KEY } from "@env";
export function Followers ({navigation}) {
    return (
      <View style={{
        backgroundColor: '#00FFE4',
      }}>
        <Text>Followers Screen</Text>
        </View>
    );
  }
  