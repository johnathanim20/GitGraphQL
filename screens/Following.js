import React, { useState, useEffect} from 'react';
import { Text, View, Button } from 'react-native';
import {parseProfile} from '../parsers/parseProfile.js';
import { API_KEY } from "@env";
export function Following ({navigation}) {
    return (
      <View style={{
        backgroundColor: '#00FFE4',
      }}>
        <Text>Following Screen</Text>
        </View>
    );
  }
  