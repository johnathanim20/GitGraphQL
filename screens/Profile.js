import React, { useState, useEffect} from 'react';
import { Text, View, Button } from 'react-native';
import {parseProfile} from '../parsers/parseProfile.js';
import { API_KEY } from "@env";

export function Profile ({navigation}) {
  var parsed;
  const [data, setData] = useState(null);
  useEffect(() => {
    const postMethod = {
      method: 'POST',
      headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
          "Authorization": `Bearer ${API_KEY}`
      },
      body: JSON.stringify({query: `{ 
          user(login:"johnathanim20") {
            avatarUrl
            name
            login
            bio
            websiteUrl
            email
            createdAt
            followers{
              totalCount
            }
            following{
              totalCount
            }
            repositories(privacy:PUBLIC, first:100){
              totalCount
              nodes{
                name
              }
            }
          }
        }`})
    };
    fetch('https://api.github.com/graphql', postMethod)
    .then(res=> {
      if(!res.ok) {
        throw Error("ERROR");
      }
      return res.json();
    })
    .then(response => {
      console.log(response);
      parsed = new parseProfile(response);
      setData(parsed);
    })
    .catch(error => {
      console.error(error);
    });
  }, []);
  console.log(data);
  return (
    <View style={{
      backgroundColor: '#00FFE4',
    }}>
        {data && <Text>Avatar URL: {data.avatar_url}</Text>}
        {data &&<Text>Name : {data.name}</Text>}
        {data &&<Text>Username : {data.username}</Text>}
        {data &&<Text>Bio : {data.bio}</Text>}
        {data &&<Text>Website : {data.website}</Text>}
        {data &&<Text>Email : {data.email}</Text>}
        {data &&<Text>Public Repo Count: {data.repositories} </Text>}
        <Button onPress = {() => navigation.navigate('Repository')}
        title = "View Repositories"></Button>
        {data && <Text>Follower Count:{data.followers}  </Text>}
        <Button
        onPress = {() => navigation.navigate('Home')}
        title = "View Followers"></Button>
        {data &&<Text>Following Count: {data.following} </Text>}
        <Button
        onPress = {() => navigation.navigate('Home')}
        title = "View Following"></Button>
        {data &&<Text>Profile Creation Date : {data.created_at}</Text>}
      </View>
  );
}

