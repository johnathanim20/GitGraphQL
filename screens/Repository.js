import React, { useState, useEffect} from 'react';
import { DataTable } from 'react-native-paper';
import { StyleSheet, Text, View, Button } from 'react-native';
import {parseRepository} from '../parsers/parseRepository.js'
import { API_KEY } from "@env";

export function Repository () {
  var parsed;
  const [data, setData] = useState(null);
  useEffect(() => {
    const postMethod = {
      method: 'POST',
      headers: {
          Accept: 'application/json',
          'Content-Type': 'application/json',
          "Authorization": `Bearer ${API_KEY}`
      },
      body: JSON.stringify({query: `{ 
          user(login:"johnathanim20") {
            repositories(privacy:PUBLIC, first:100){
              nodes{
                name
                owner {
                  login
                }
                description
              }
            }
          }
        }`})
    };
    fetch('https://api.github.com/graphql', postMethod)
    .then(res=> {
      if(!res.ok) {
        throw Error("ERROR");
      }
      return res.json();
    })
    .then(response => {
      console.log(response);
      parsed = new parseRepository(response);
      setData(parsed);
    })
    .catch(error => {
      console.error(error);
    });
  }, []);
  
  return (
    <View style={{
      container: {
        paddingTop: 0,
        paddingHorizontal: 0,
      }
    }}>
      <DataTable>
        <DataTable.Header>
          <DataTable.Title style = {{flex:1}}>Repo Name</DataTable.Title>
          <DataTable.Title style = {{flex:1.5}}>Username</DataTable.Title>
          <DataTable.Title style = {{flex:1}}>Description</DataTable.Title>
        </DataTable.Header>
        {
          data && data.name.map((entry, name) => {
              return(
                <DataTable.Row key = {name}>
                  <DataTable.Cell style = {{flex:1}}>{entry.name}</DataTable.Cell>
                  <DataTable.Cell style = {{flex:2}}>{entry.owner.login}</DataTable.Cell>
                  <Text style = {{flex:2, justifyContent: 'center'}}>{entry.description}</Text>
                </DataTable.Row>
              )
          })
        }     
      </DataTable>
    </View>
  );
}
