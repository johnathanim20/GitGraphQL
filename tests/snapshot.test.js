import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Profile} from '../screens/Profile.js';
import {Repository} from '../screens/Repository.js';
import {Followers} from '../screens/Followers.js'
import {Following} from '../screens/Following.js'

beforeEach(() => {
    jest.useFakeTimers()
})

it("snapshot1", () => {
    
    const snap = renderer.create(
        (<Profile/>)
    ).toJSON();

    expect(snap).toMatchSnapshot();
});

it("snapshot2", () => {
    
    const snap = renderer.create(
        (<Repository/>)
    ).toJSON();
    expect(snap).toMatchSnapshot();
});
it("snapshot3", () => {
    
    const snap = renderer.create(
        (<Following/>)
    ).toJSON();
    expect(snap).toMatchSnapshot();
});
it("snapshot4", () => {
    
    const snap = renderer.create(
        (<Followers/>)
    ).toJSON();
    expect(snap).toMatchSnapshot();
});