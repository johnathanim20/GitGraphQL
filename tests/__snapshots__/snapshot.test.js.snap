// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot1 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#00FFE4",
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        View Repositories
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        View Followers
      </Text>
    </View>
  </View>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        View Following
      </Text>
    </View>
  </View>
</View>
`;

exports[`snapshot2 1`] = `
<View
  style={
    Object {
      "container": Object {
        "paddingHorizontal": 0,
        "paddingTop": 0,
      },
    }
  }
>
  <View
    style={
      Array [
        Object {
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "height": 48,
            "paddingHorizontal": 16,
          },
          Object {
            "borderBottomColor": "rgba(0, 0, 0, 0.12)",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignContent": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingVertical": 12,
            },
            undefined,
            Object {
              "flex": 1,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "fontSize": 12,
                  "fontWeight": "500",
                  "height": 24,
                  "lineHeight": 24,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.6)",
                },
              ],
            ]
          }
        >
          Repo Name
        </Text>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignContent": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingVertical": 12,
            },
            undefined,
            Object {
              "flex": 1.5,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "fontSize": 12,
                  "fontWeight": "500",
                  "height": 24,
                  "lineHeight": 24,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.6)",
                },
              ],
            ]
          }
        >
          Username
        </Text>
      </View>
      <View
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "alignContent": "center",
              "flex": 1,
              "flexDirection": "row",
              "paddingVertical": 12,
            },
            undefined,
            Object {
              "flex": 1,
            },
          ]
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
              },
              Object {
                "textAlign": "left",
              },
              Array [
                Object {
                  "alignItems": "center",
                  "fontSize": 12,
                  "fontWeight": "500",
                  "height": 24,
                  "lineHeight": 24,
                },
                Object {
                  "color": "rgba(0, 0, 0, 0.6)",
                },
              ],
            ]
          }
        >
          Description
        </Text>
      </View>
    </View>
  </View>
</View>
`;

exports[`snapshot3 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#00FFE4",
    }
  }
>
  <Text>
    Following Screen
  </Text>
</View>
`;

exports[`snapshot4 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#00FFE4",
    }
  }
>
  <Text>
    Followers Screen
  </Text>
</View>
`;
